src/sagemaker/session.py

```diff
@@ -4352,6 +4352,28 @@ def update_endpoint(self, endpoint_name, endpoint_config_name, wait=True):
             self.wait_for_endpoint(endpoint_name)
         return endpoint_name
 
+    def is_ic_based_endpoint(self, endpoint_name):
+        """Returns True if endpoint is inference component based, False otherwise.
+
+        An endpoint is inference component based if and only if the associated endpoint config
+        has a role associated with it and a single production variant with no ``ModelName`` field.
```

Comment:
sanity-check: the docstring description doesn't align with the checks you are performing on line 4372 (you also check the execution role ARN for the endpoint config). Is that intentional?

Reply:
don't I mention `... if the associated endpoint config has a role associated with it ..." or am I missing something?